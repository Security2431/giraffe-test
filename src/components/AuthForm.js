import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

class AuthForm extends React.Component {
  static propTypes = {
    isAuthenticated: PropTypes.bool.isRequired,
    activeUser: PropTypes.string.isRequired,
    logout: PropTypes.func.isRequired,
  };

  state = {
    username: {
      value: '',
      isValid: true,
    },
    password: {
      value: '',
      isValid: true,
    },
  };

  handleInputChange = (event) => {
    event.persist();
    const { name, value } = event.target;

    this.setState(prevState => ({
      [name]: {
        ...prevState[name],
        value,
      },
    }));
  };

  handleSubmit = (event) => {
    event.preventDefault();

    const { username, password } = this.state;

    // eslint-disable-next-line
    this.props.onSubmit(username.value, password.value);
  };

  render() {
    const { isAuthenticated, activeUser, logout } = this.props;
    const { username, password } = this.state;

    return (
      <React.Fragment>
        {!isAuthenticated ? (
          <form className="form-inline" onSubmit={this.handleSubmit}>
            <div className="row">
              <div className="col">
                <div className="form-group">
                  <input
                    className="form-control"
                    type="text"
                    name="username"
                    placeholder="Username"
                    autoComplete="username"
                    value={username.value}
                    onChange={this.handleInputChange}
                    required
                  />
                </div>
              </div>
              <div className="col">
                <div className="form-group">
                  <input
                    className="form-control"
                    type="password"
                    name="password"
                    placeholder="********"
                    value={password.value}
                    autoComplete="password"
                    onChange={this.handleInputChange}
                    required
                  />
                </div>
              </div>
              <div className="col">
                <button className="btn btn-primary" type="submit">
                  Sign in
                </button>
              </div>
            </div>
          </form>
        ) : (
          <div className="row align-items-center">
            <div className="col">
              <b>{activeUser}</b>
            </div>
            <div className="col">
              <button className="btn btn-link" type="button" onClick={logout}>
                <Link to="/">LogÂ out</Link>
              </button>
            </div>
          </div>
        )}
      </React.Fragment>
    );
  }
}

export default AuthForm;
