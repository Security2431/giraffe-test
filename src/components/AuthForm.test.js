/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import AuthForm from './AuthForm';

// Return a fixed timestamp when moment().fromNow() is called
jest.mock('moment', () => () => ({ fromNow: () => '2 days ago' }));

const mockProps = {
  isAuthenticated: true,
  activeUser: 'Admin',
  logout: jest.fn(),
};

describe('<AuthForm />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <AuthForm {...mockProps} />
      </MemoryRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <AuthForm {...mockProps} />
        </MemoryRouter>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
