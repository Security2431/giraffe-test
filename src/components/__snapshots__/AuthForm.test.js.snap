// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AuthForm /> renders correctly 1`] = `
<div
  className="row align-items-center"
>
  <div
    className="col"
  >
    <b>
      Admin
    </b>
  </div>
  <div
    className="col"
  >
    <button
      className="btn btn-link"
      onClick={[MockFunction]}
      type="button"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        LogÂ out
      </a>
    </button>
  </div>
</div>
`;
